bin/live/pycbc_live_collate_triggers

```diff
@@ -0,0 +1,333 @@
+    start_date = datetime.strptime(args.start_date, '%Y-%m-%d').date()
+    end_date = datetime.strptime(args.end_date, '%Y-%m-%d').date()
+
+    # Create the days array
+    days = []
+    delta = timedelta(days=1)
+
+    # Loop through the days and add each date to the list
+    current_date = start_date
+    while current_date <= end_date:
+        days.append(current_date)
+        current_date += delta
+
+    # Convert to strings in a list
+    days = [str(date).replace('-', '_') for date in days]
+    num_days = str(len(days))
+
+    initial_trigger_files = [
+        os.path.join(args.trigger_dir, day, trigger_file)
+        for day in days
+        for trigger_file in os.listdir(os.path.join(args.trigger_dir, day))
+    ]
+
+if args.trigger_file_method == 'start-num-days':
+    if not args.start_date:
+        parser.error('Please provide a start date.')
+    if not args.num_days:
+        parser.error('Please provide a number of days.')
+    if not args.trigger_dir:
+        parser.error('Please provide a directory containing trigger files.')
+
+    # Convert dates to datetime to get the days array
+    start_date = datetime.strptime(args.start_date, '%Y-%m-%d').date()
+    num_days = timedelta(days=args.num_days - 1)
+    end_date = start_date + num_days
+
+    # Create the days array
+    days = []
+    delta = timedelta(days=1)
+
+    # Loop through the days and add each date to the list
+    current_date = start_date
+    while current_date <= end_date:
+        days.append(current_date)
+        current_date += delta
+
+    # Convert to strings in a list
+    days = [str(date).replace('-', '_') for date in days]
+    num_days = str(len(days))
+
+    initial_trigger_files = [
+        os.path.join(args.trigger_dir, day, trigger_file)
+        for day in days
+        for trigger_file in os.listdir(os.path.join(args.trigger_dir, day))
+    ]
+
+if args.trigger_file_method == 'gps-start-end-time':
+    if not args.gps_start_time:
+        parser.error('Please provide a GPS start time.')
+    if not args.gps_end_time:
+        parser.error('Please provide a GPS end time.')
+    if not args.trigger_dir:
+        parser.error('Please provide a directory containing trigger files.')
+
+    initial_trigger_files = [
+        l for l in glob.glob(args.trigger_dir + '/*/*-Live-*.hdf', recursive=True)
+        if float(l.split('/')[-1].split("-")[2]) > args.gps_start_time and float(l.split('/')[-1].split("-")[2]) < args.gps_end_time
+    ]
+
+logging.info(f" {len(initial_trigger_files)} files found")
+
+# Check if all files found are in the correct format and are trigger files
+ifo_permutations = [''.join(p) for p in itertools.permutations(args.ifos)]
+ifo_prefix_set = set(ifo_permutations)
+
+trigger_files = []
+for source_file in initial_trigger_files:
+    trigger_file = os.path.basename(source_file)
+    if any(trigger_file.startswith(prefix + '-Live') for prefix in ifo_prefix_set):
+        trigger_files.append(source_file)
```

Comment:
At some point here you should also check that they are HDF5 files (extension `.hdf` or `.h5`) though this is still not guaranteed to give you only the files you are looking for.

Reply:
The reason here is that if you end up with a PNG file (for example), the code will throw an exception when you try to load it with `h5py`.